Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp

```diff
@@ -240,15 +252,24 @@ void BreakpointWidget::Update()
 
 void BreakpointWidget::OnDelete()
 {
-  if (m_table->selectedItems().empty())
+  const auto selected_items = m_table->selectedItems();
+  if (selected_items.empty())
     return;
 
-  auto address = m_table->selectedItems()[0]->data(Qt::UserRole).toUInt();
+  const auto item = selected_items.constFirst();
+  const auto address = item->data(ADDRESS_ROLE).toUInt();
+  const bool is_memcheck = item->data(IS_MEMCHECK_ROLE).toBool();
 
-  PowerPC::breakpoints.Remove(address);
-  Settings::Instance().blockSignals(true);
-  PowerPC::memchecks.Remove(address);
-  Settings::Instance().blockSignals(false);
+  if (is_memcheck)
+  {
+    Settings::Instance().blockSignals(true);
```

Comment:
I realise this was already written like this, but using a QSignalBlocker would be cleaner here. Idea for a follow-up PR :)